tls/s2n_psk.h

```diff
@@ -50,7 +62,9 @@ struct s2n_psk_parameters {
     s2n_psk_key_exchange_mode psk_ke_mode;
 };
 
-int s2n_psk_init(struct s2n_psk *psk, s2n_psk_type type);
+S2N_RESULT s2n_connection_set_psks(struct s2n_connection *conn, struct s2n_pre_shared_key *psk_vec, size_t *psk_vec_length);
```

Comment:
We could make use of `s2n_blob` to represent psk_identity  and `s2n_stuffer` to obtain the psk_identities list.

Reply:
Also, for APIs we generally use `int` as the return type and not `S2N_RESULT`.

Reply:
As above, we can't give internal concepts like stuffers or blobs to our customers to use, they wouldn't know how to use it. I think you're right about not using S2N_RESULT though.